lambdas/call-dcmaw-async-cri/src/test/java/uk/gov/di/ipv/core/calldcmawasynccri/CallDcmawAsyncCriHandlerTest.java

```diff
@@ -0,0 +1,167 @@
+                    .clientOAuthSessionId(TEST_OAUTH_STATE)
+                    .govukSigninJourneyId("a-govuk-journey-id")
+                    .build();
+    private static final ProcessRequest input =
+            ProcessRequest.processRequestBuilder()
+                    .ipvSessionId("a-session-id")
+                    .ipAddress("an-ip-address")
+                    .clientOAuthSessionId("an-oauth-session-id")
+                    .journey("a-journey")
+                    .lambdaInput(Map.of("journeyType", "ipv"))
+                    .build();
+    public static final String JOURNEY_ENHANCED_VERIFICATION = "/journey/enhanced-verification";
+    @Mock private Context mockContext;
+    @Mock private ConfigService mockConfigService;
+    @Mock private IpvSessionService mockIpvSessionService;
+    @Mock private ClientOAuthSessionDetailsService mockClientOAuthSessionDetailsService;
+    @Mock private DcmawAsyncCriService mockDcmawAsyncCriService;
+    @Mock private CriStoringService mockCriStoringService;
+    @Mock private IpvSessionItem mockIpvSessionItem;
+    @Mock private VerifiableCredentialResponse mockVerifiableCredentialResponse;
+    @InjectMocks private CallDcmawAsyncCriHandler callDcmawAsyncCriHandler;
+
+    @BeforeEach
+    public void setUp() {
+        mockIpvSessionItem.setIpvSessionId("a-session-id");
+    }
+
+    @Test
+    void handleRequestShouldCallDcmawCriAndReturnJourneyNextIfNoBreachingCiReceived()
+            throws Exception {
+        when(mockIpvSessionService.getIpvSession("a-session-id")).thenReturn(mockIpvSessionItem);
+        when(mockClientOAuthSessionDetailsService.getClientOAuthSession(any()))
+                .thenReturn(clientOAuthSessionItem);
+        when(mockVerifiableCredentialResponse.getUserId()).thenReturn(TEST_USER_ID);
+        when(mockVerifiableCredentialResponse.getCredentialStatus())
+                .thenReturn(VerifiableCredentialStatus.PENDING);
+        when(mockDcmawAsyncCriService.startDcmawAsyncSession(
+                        any(String.class),
+                        any(ClientOAuthSessionItem.class),
+                        any(IpvSessionItem.class)))
+                .thenReturn(mockVerifiableCredentialResponse);
+
+        Map<String, Object> lambdaResult =
+                callDcmawAsyncCriHandler.handleRequest(input, mockContext);
+
+        verify(mockCriStoringService)
+                .storeCriResponse(
+                        eq(input),
+                        eq(DCMAW_ASYNC_CRI),
+                        any(String.class),
+                        eq(clientOAuthSessionItem));
+
+        InOrder inOrder = inOrder(mockIpvSessionService);
+        inOrder.verify(mockIpvSessionService).updateIpvSession(mockIpvSessionItem);
+        inOrder.verifyNoMoreInteractions();
+
+        assertEquals("/journey/next", lambdaResult.get("journey"));
+    }
+
+    @Test
+    void handleRequestShouldReturnJourneyErrorResponseIfMissingIpvSessionId() {
```

Comment:
This test seems to be testing the handling of `HttpResponseExceptionWithErrorBody` exceptions and it's just convenient that `RequestHelper.getIpvSessionId()` throws one of these when the session ID is missing. Perhaps this test needs renaming.